### PR TITLE
Feature: collect pipeline central, postqc and product release logs from BAM_basecalls folder

### PR DESCRIPTION
**Pull request by marcomoscasgr**
`Central pipeline logs`, `post qc review JSON files` and `product release yml` files are under `BAM_basecalls` folder and contain the pipeline steps, commands and RoboQC config chosen. With this PR they will be archived in iRODS.

### CODE REVIEW

File: t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm [hunk 1]
```diff
@@ -71,6 +74,43 @@ sub publish_logs : Test(4) {
...
+  Readonly::Scalar my $LOGS_COUNT => 43;
+  my @log_files =
+    ('_software_npg_20241107_bin_npg_pipeline_central_17550_20241101-132705-1566962201.definitions.json',
+      '_software_npg_20241107_bin_npg_pipeline_post_qc_review_17550_20241104-124525-1544617117.definitions.json',
+      '_software_npg_20241107_bin_npg_pipeline_central_17550_20241101-132705-1566962201.log',
+      '_software_npg_20241107_bin_npg_pipeline_post_qc_review_17550_20241104-124525-1544617117.log',
+      'product_release_20241101-132705-1247165951.yml',
+      'product_release_20241104-124525-1500421497.yml');
+  my $temp_dir = tempdir(CLEANUP => 1);
+  my $ref_name = '151211_HX3_18448_B_HHH55CCXX';
```
Comment (mgcam) [inline t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm:88]: $ref_name looks suspiciously similar to a short name for reference_name. Probably change to $rf_name or similar?

File: t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm [hunk 2]
```diff
@@ -71,6 +74,43 @@ sub publish_logs : Test(4) {
...
+  dircopy("t/data/illumina/sequence/$ref_name", $runfolder_path3);
+  for my $log (@log_files) {
+    open my $handle, '>', File::Spec->catfile($bam_basecall_folder3, $log)
+      or die "error while creating test logs $log";
+    close $handle or die "error while closing test logs $log";
+  }
+  my $pub3 = WTSI::NPG::HTS::Illumina::LogPublisher->new
+    (irods           => $irods,
+     runfolder_path  => $runfolder_path3,
+		 id_run          => 18448,
```
Comment (mgcam) [inline t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm:101]: Line 101 is still misaligned

File: t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm [hunk 3]
```diff
@@ -7,6 +7,9 @@ use Carp;
 use English qw[-no_match_vars];
 use Log::Log4perl;
 use Test::More;
+use File::Temp qw(tempdir);
+use File::Copy::Recursive qw(dircopy);
+use File::Spec;
```
Comment (mgcam) [inline t/lib/WTSI/NPG/HTS/Illumina/LogPublisherTest.pm:12]: If any of these three packages are not in the 'requires' or 'build_requires' sections of Build.PL, please add them to the 'build_requires' section.

File: lib/WTSI/NPG/HTS/Illumina/LogPublisher.pm [hunk 4]
```diff
@@ -26,6 +27,8 @@ our $VERSION = '';
 our $DEFAULT_ROOT_COLL = '/seq';
 our $DEFAULT_LOG_COLL  = 'log';
 
+Readonly::Scalar my $BAM_BASECALLS_DEPTH => 4;
```
Comment (mgcam) [inline lib/WTSI/NPG/HTS/Illumina/LogPublisher.pm:30]: To be consistend, please update definitions of constants om lines 27-28 to

Readonly::Scalar my $DEFAULT_ROOT_COLL => '/seq';
Readonly::Scalar my $DEFAULT_LOG_COLL  => 'log';

**Comment by mgcam** [comment]
Please update the Copyright year in the Perl package you were changing.

**Review by mgcam — CHANGES_REQUESTED** [review]
Looks good. Please address minor concerns

File: lib/WTSI/NPG/HTS/Illumina/LogPublisher.pm (hunk 4 above)
Comment (marcomoscasgr) [reply lib/WTSI/NPG/HTS/Illumina/LogPublisher.pm:30]: Should I leave `our` keyword for these variables?

Comment (mgcam) [reply lib/WTSI/NPG/HTS/Illumina/LogPublisher.pm:30]: No, they are not used anywhere. You can check by searching the code of this packege. Classess from this package are not used anywhere else in NPG code.

**Pull request merged**